# Random Forest Classifier carbon, silicon, and nitrogen isotopes

In [9]:
#Imports
import joblib
import pandas as pd

from sklearn import tree
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder, MinMaxScaler


In [10]:
# Read csv file
C_Si_N = pd.read_csv('presolargrains_C_Si_N.csv')
C_Si_N.head()

Type  carbon_12_13  nitrogen_14_15  silicon_29_28  silicon_30_28
0    X         74.30           207.4         -685.0         -520.0
1    X       1581.00           116.0         -684.0         -490.0
2    X        234.75           187.2         -683.0         -501.0
3    X        455.00           140.0         -662.0         -770.0
4    X        823.00            44.0         -658.0         -234.0

In [11]:
# Set up data and feature names; display dataframe dropping columns
data = C_Si_N.drop("Type", axis=1)
feature_names = data.columns
data.head()

carbon_12_13  nitrogen_14_15  silicon_29_28  silicon_30_28
0         74.30           207.4         -685.0         -520.0
1       1581.00           116.0         -684.0         -490.0
2        234.75           187.2         -683.0         -501.0
3        455.00           140.0         -662.0         -770.0
4        823.00            44.0         -658.0         -234.0

In [12]:
target = C_Si_N['Type']

In [13]:
# Separate data into train and test buckets
X_train, X_test, y_train, y_test = train_test_split(data, target, random_state=1)

## Scale the data using MinMaxScalar

In [14]:
X_scaler = MinMaxScaler().fit(X_train)
X_train_scaled = X_scaler.transform(X_train)
X_test_scaled = X_scaler.transform(X_test)

## Train the model; run the test score

In [15]:
# Random Forest Classifier
rf = RandomForestClassifier(n_estimators=200)
rf = rf.fit(X_train_scaled, y_train)
rf.score(X_test_scaled, y_test)

0.9693251533742331

In [16]:
# Save fitted model to file
filename = 'RandomForestCls_C_Si_N.sav'
joblib.dump(rf, filename)

['RandomForestCls_C_Si_N.sav']

## Summary:

### Random Forest Classifier carbon, silicon, and nitrogen:  96.9%